File: app/src/main/java/com/lineclient/home/homelineclient/fragment/BaseFragment.java
package com.lineclient.home.homelineclient.fragment;

import android.os.Bundle;
import android.support.annotation.Nullable;
import android.support.v4.app.Fragment;

import com.lineclient.home.homelineclient.activity.BaseActivity;

public class BaseFragment extends Fragment {

    private BaseActivity baseActivity;

    @Override
    public void onCreate(@Nullable Bundle savedInstanceState) {
        super.onCreate(savedInstanceState);
        baseActivity = (BaseActivity) this.getActivity();
    }

    public void showLongToast(String text) {
        baseActivity.showLongToast(text);
    }

    public void showShortToast(String text) {
        baseActivity.showShortToast(text);
    }

    public void showLoading() {
        baseActivity.showLoading();
    }

    public void closeLoading() {
        baseActivity.closeLoading();
    }

}

File: app/src/main/java/com/lineclient/home/homelineclient/fragment/BaseFragment.kt
package com.lineclient.home.homelineclient.fragment

import android.os.Bundle
import android.support.v4.app.Fragment
import com.lineclient.home.homelineclient.activity.BaseActivity

open class BaseFragment : Fragment() {
    private var baseActivity: BaseActivity? = null
    override fun onCreate(savedInstanceState: Bundle?) {
        super.onCreate(savedInstanceState)
        baseActivity = this.activity as BaseActivity?
    }

    fun showLongToast(text: String?) {
        baseActivity?.showLongToast(text)
    }

    fun showShortToast(text: String?) {
        baseActivity?.showShortToast(text)
    }

    fun showLoading() {
        baseActivity?.showLoading()
    }

    fun closeLoading() {
        baseActivity?.closeLoading()
    }
}